# Logistic Regression and Binary Logit Model
* name: **Connor Oberhofer**
* ref: https://www.statsmodels.org/dev/examples/notebooks/generated/discrete_choice_overview.html
* ref: https://github.com/woerman/ResEcon703

### Short-cut for Jupyter-notebook
* M change it as text
* B put a code cell

In [ ]:
# run this code to call package
import numpy as np
import pandas as pd
import statsmodels.api as sm

/usr/local/lib/python3.7/dist-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  import pandas.util.testing as tm


### Data load (in the package)
* Load data from Spector and Mazzeo (1980). Examples follow Greene’s Econometric Analysis Ch. 21 (5th Edition).

### Working directory

In [ ]:
# just run to check your working directory
# your raw file should be in this working directory
import os
os.getcwd()

'/content'

In [ ]:
# just run this code to upload the raw file from your working directory 
df = pd.read_csv ('logit_data.csv')
df

air_conditioning  cost_system  cost_operating  ...  income  residents  city
0                   0          513             247  ...      47          2     1
1                   0          578             138  ...      64          1     1
2                   1          658             171  ...      86          2     1
3                   0          615             198  ...      49          2     1
4                   0          515             165  ...      56          1     1
..                ...          ...             ...  ...     ...        ...   ...
595                 0          663             162  ...      69          1     3
596                 0          626             162  ...      47          1     3
597                 1          511             149  ...      48          3     3
598                 1          528             117  ...      43          3     3
599                 1          619             260  ...      70          3     3

[600 rows x 8 columns]

In [ ]:
print(df.shape)
print(type(df))

(600, 8)
<class 'pandas.core.frame.DataFrame'>


# Meaning of variables
* An indicator if they purchase the air conditioner, TRUE/FALSE
* The purchase price of the air conditioner
* The annual operating cost of the air conditioner, dollar per year
* The household’s electricity price cents per kWh
* The size of the household’s apartment , square feet
* The household’s annual income ,$1000s
* The number of residents in the household ,people
* An indicator for the household’s city 1, 2, or 3

### Random Number generation
* Please put some number that you like in **random.seed( the number that you like)** commend 

In [ ]:
import random 
random.seed(55) # <----place any number inside of random.seed()
your_random_number = random.randrange(1, 101)
print(your_random_number)

12


### Creating new variable based on your random number

In [ ]:
# creates new price variable using purchasing price of AC x random number
df['new_price'] = df['cost_system']*your_random_number

In [ ]:
# now, you can see the last column has new variable called "new_price"
df

air_conditioning  cost_system  cost_operating  ...  residents  city  new_price
0                   0          513             247  ...          2     1       6156
1                   0          578             138  ...          1     1       6936
2                   1          658             171  ...          2     1       7896
3                   0          615             198  ...          2     1       7380
4                   0          515             165  ...          1     1       6180
..                ...          ...             ...  ...        ...   ...        ...
595                 0          663             162  ...          1     3       7956
596                 0          626             162  ...          1     3       7512
597                 1          511             149  ...          3     3       6132
598                 1          528             117  ...          3     3       6336
599                 1          619             260  ...          3     3       7428

[600 rows x 9 columns]

In [ ]:
# just run code for summary table
df.describe()

air_conditioning  cost_system  ...        city    new_price
count        600.000000   600.000000  ...  600.000000   600.000000
mean           0.568333   594.190000  ...    2.000000  7130.280000
std            0.495722    57.865804  ...    0.817178   694.389653
min            0.000000   490.000000  ...    1.000000  5880.000000
25%            0.000000   546.000000  ...    1.000000  6552.000000
50%            1.000000   592.500000  ...    2.000000  7110.000000
75%            1.000000   646.000000  ...    3.000000  7752.000000
max            1.000000   693.000000  ...    3.000000  8316.000000

[8 rows x 9 columns]

### Summary Statistics:
*  what is the maximum value of the new_price variable based on the following summary table?

 $8316



### Column names

In [ ]:
df.columns

Index(['air_conditioning', 'cost_system', 'cost_operating', 'elec_price',
       'square_feet', 'income', 'residents', 'city', 'new_price'],
      dtype='object')

### Logistic Regression


In [ ]:
Y = df[['air_conditioning']]
Y.head()

air_conditioning
0                 0
1                 0
2                 1
3                 0
4                 0

In [ ]:
X = df[['new_price']]
X.head()

new_price
0       6156
1       6936
2       7896
3       7380
4       6180

# Binary logit model
* intro level tutorial in R: https://www.princeton.edu/~otorres/LogitR101.pdf
* goed reference lecture and slide: https://github.com/woerman/ResEcon703

In [ ]:
# Just run this code
logit_model = sm.Logit(Y, X)
logit_result = logit_model.fit(disp=0)
print("Parameters: ", logit_result.params)

Parameters:  new_price    0.000035
dtype: float64


In [ ]:
# Just run this code
print(logit_result.summary())

                           Logit Regression Results                           
Dep. Variable:       air_conditioning   No. Observations:                  600
Model:                          Logit   Df Residuals:                      599
Method:                           MLE   Df Model:                            0
Date:                Thu, 27 Jan 2022   Pseudo R-squ.:               -0.002087
Time:                        22:21:00   Log-Likelihood:                -411.12
converged:                       True   LL-Null:                       -410.27
Covariance Type:            nonrobust   LLR p-value:                       nan
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
new_price   3.533e-05   1.15e-05      3.074      0.002    1.28e-05    5.79e-05


### Dependent Variable:
air_conditioning   

### Independent Variable:
new_price

### Coefficient value of the independent variable:
3.533e-05 

### Is the coefficient value of the independent variable statistically significant? 
Yes, at 0.002.

### Is the effect of the independent variable positive or negative on the dependent variable? 
positive

### Log-Likelihoodvalue: 
-411.12

# Marginal Effect of Binary Logit

In [ ]:
# Just run
margeff = logit_result.get_margeff()
print(margeff.summary())

        Logit Marginal Effects       
Dep. Variable:       air_conditioning
Method:                          dydx
At:                           overall
                dy/dx    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
new_price   8.692e-06   2.74e-06      3.175      0.001    3.33e-06    1.41e-05


### Marginal effect of the independent variable:
8.692e-06

### Is the marginal effect of independent variables statistically significant?
Yes, at P value is 0.001.

### What is the meaning of marginal effect on the independent variables?
The meaning of marginal effect on new_price is that it is a measure of the instantaneous effect that a change in air_conditioning has on the predicted probability